tests/platform_tests/test_intf_fec.py

```diff
@@ -147,3 +147,58 @@ def test_verify_fec_stats_counters(duthosts, enum_rand_one_per_hwsku_frontend_ho
         if fec_symbol_err_int > fec_corr_int:
             pytest.fail("FEC symbol errors:{} are higher than FEC correctable errors:{} for interface {}"
                         .format(intf_name, fec_symbol_err_int, fec_corr_int))
+
+
+def test_verify_fec_histogram(duthosts, enum_rand_one_per_hwsku_frontend_hostname,
+                              enum_frontend_asic_index, conn_graph_facts):
+    """
+    @Summary: Verify the FEC histogram is valid and check for errors
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+
+    logging.info("Get output of 'show interface status'")
+    intf_status = duthost.show_and_parse("show interface status")
+
+    for intf in intf_status:
+        intf_name = intf['interface']
+        sfp_presence = duthost.show_and_parse("sudo sfpshow presence -p {}".format(intf_name))
+        if sfp_presence:
+            presence = sfp_presence[0].get('presence', '').lower()
+            oper = intf.get('oper', '').lower()
+            speed = intf.get('speed', '')
+
+            # Skip interfaces that are not up or have no SFP module present or for unsupported speeds
+            if presence == "not present" or oper != "up" or speed not in SUPPORTED_SPEEDS:
+                continue
```

Comment:
@vvolam can we add a log here for skipping also print the presence, oper and speed?